refact(exp): Modify the label name for jobs to distinguish with multiple same jobs

w3aman opened the pull request:
Signed-off-by: Aman Gupta <aman.gupta@mayadata.io>

<!--  Thanks for sending a pull request!  Here are some tips for you -->

**What this PR does / why we need it**:
- When we run zv-property-modify job multiple time simultaneously, due to same label in daemonset pod each time, we end up getting multiple daemonset pods on a single node and was not able to distinguish which one is for this running experiment

nsathyaseelan commented on experiments/functional/zfs-LocalPV/zv-property-runtime-modify/run_litmus_test.yml:
```diff
@@ -26,6 +26,9 @@ spec:
           - name: APP_LABEL  ## Application label
             value: ''
 
+          - name: FILE_SYSTEM_TYPE
```
@w3aman  can you provide what are all the supported values here

w3aman commented:
Logs for the job.
```
PLAYBOOK: test.yml *************************************************************
1 plays in ./experiments/functional/zfs-LocalPV/zv-property-runtime-modify/test.yml

PLAY [localhost] ***************************************************************
2020-07-02T06:43:10.026744 (delta: 0.046156)         elapsed: 0.046156 ******** 
=============================================================================== 
META: ran handlers

TASK [include_tasks] ***********************************************************
task path: /experiments/functional/zfs-LocalPV/zv-property-runtime-modify/test.yml:12
2020-07-02T06:43:10.038851 (delta: 0.012068)         elapsed: 0.058263 ******** 
included: /utils/fcm/create_testname.yml for 127.0.0.1

TASK [Record test instance/run ID] *********************************************
task path: /utils/fcm/create_testname.yml:3
2020-07-02T06:43:10.078871 (delta: 0.040004)         elapsed: 0.098283 ******** 
skipping: [127.0.0.1] => {"changed": false, "skip_reason": "Conditional result was False"}

TASK [Construct testname appended with runID] **********************************
task path: /utils/fcm/create_testname.yml:7
2020-07-02T06:43:10.148885 (delta: 0.069978)         elapsed: 0.168297 ******** 
skipping: [127.0.0.1] => {"changed": false, "skip_reason": "Conditional result was False"}

TASK [include_tasks] ***********************************************************
task path: /experiments/functional/zfs-LocalPV/zv-property-runtime-modify/test.yml:15
2020-07-02T06:43:10.231483 (delta: 0.082563)         elapsed: 0.250895 ******** 
included: /utils/fcm/update_litmus_result_resource.yml for 127.0.0.1

TASK [Generate the litmus result CR to reflect SOT (Start of Test)] ************
task path: /utils/fcm/update_litmus_result_resource.yml:3
2020-07-02T06:43:10.273413 (delta: 0.04189)         elapsed: 0.292825 ********* 
changed: [127.0.0.1] => {"changed": true, "checksum": "840f3b5742ea45e6fb2c6c284dbb1db24249ef3d", "dest": "./litmus-result.yaml", "gid": 0, "group": "root", "md5sum": "1c30751caa6a6f10500548c1364d9861", "mode": "0644", "owner": "root", "size": 429, "src": "/root/.ansible/tmp/ansible-tmp-1593672190.31-103891123897713/source", "state": "file", "uid": 0}

TASK [Analyze the cr yaml] *****************************************************
task path: /utils/fcm/update_litmus_result_resource.yml:14
2020-07-02T06:43:10.711191 (delta: 0.437751)         elapsed: 0.730603 ******** 
changed: [127.0.0.1] => {"changed": true, "cmd": "cat litmus-result.yaml", "delta": "0:00:00.505185", "end": "2020-07-02 06:43:11.458618", "rc": 0, "start": "2020-07-02 06:43:10.953433", "stderr": "", "stderr_lines": [], "stdout": "---\napiVersion: litmus.io/v1alpha1\nkind: LitmusResult\nmetadata:\n\n  # name of the litmus testcase\n  name: zvol-property-runtime-modify \nspec:\n\n  # holds information on the testcase\n  testMetadata:\n    app:  \n    chaostype:  \n\n  # holds the state of testcase,  manually updated by json merge patch\n  # result is the useful value today, but anticipate phase use in future \n  testStatus: \n    phase: in-progress  \n    result: none ", "stdout_lines": ["---", "apiVersion: litmus.io/v1alpha1", "kind: LitmusResult", "metadata:", "", "  # name of the litmus testcase", "  name: zvol-property-runtime-modify ", "spec:", "", "  # holds information on the testcase", "  testMetadata:", "    app:  ", "    chaostype:  ", "", "  # holds the state of testcase,  manually updated by json merge patch", "  # result is the useful value today, but anticipate phase use in future ", "  testStatus: ", "    phase: in-progress  ", "    result: none "]}

TASK [Apply the litmus result CR] **********************************************
task path: /utils/fcm/update_litmus_result_resource.yml:17
2020-07-02T06:43:11.492069 (delta: 0.780832)         elapsed: 1.511481 ******** 
changed: [127.0.0.1] => {"changed": true, "failed_when_result": false, "method": "patch", "result": {"apiVersion": "litmus.io/v1alpha1", "kind": "LitmusResult", "metadata": {"creationTimestamp": "2020-07-01T08:23:18Z", "generation": 9, "name": "zvol-property-runtime-modify", "resourceVersion": "952600", "selfLink": "/apis/litmus.io/v1alpha1/litmusresults/zvol-property-runtime-modify", "uid": "609f3a7a-2d29-472e-8502-2a810730f1db"}, "spec": {"testMetadata": {}, "testStatus": {"phase": "in-progress", "result": "none"}}}}

TASK [Generate the litmus result CR to reflect EOT (End of Test)] **************
task path: /utils/fcm/update_litmus_result_resource.yml:27
2020-07-02T06:43:12.300987 (delta: 0.808884)         elapsed: 2.320399 ******** 
skipping: [127.0.0.1] => {"changed": false, "skip_reason": "Conditional result was False"}

TASK [Analyze the cr yaml] *****************************************************
task path: /utils/fcm/update_litmus_result_resource.yml:38
2020-07-02T06:43:12.325725 (delta: 0.024706)         elapsed: 2.345137 ******** 
skipping: [127.0.0.1] => {"changed": false, "skip_reason": "Conditional result was False"}

TASK [Apply the litmus result CR] **********************************************
task path: /utils/fcm/update_litmus_result_resource.yml:41
2020-07-02T06:43:12.350433 (delta: 0.024671)         elapsed: 2.369845 ******** 
skipping: [127.0.0.1] => {"changed": false, "skip_reason": "Conditional result was False"}

TASK [Update the daemonset spec template with test specific values] ************
task path: /experiments/functional/zfs-LocalPV/zv-property-runtime-modify/test.yml:20
2020-07-02T06:43:12.377819 (delta: 0.02734)         elapsed: 2.397231 ********* 
changed: [127.0.0.1] => {"changed": true, "checksum": "328b1a556407d2bb51b4b08154896306375a4e7f", "dest": "./zv_property_ds.yml", "gid": 0, "group": "root", "md5sum": "2edcc626a0a4259e2987beb1ba4a49fa", "mode": "0644", "owner": "root", "size": 899, "src": "/root/.ansible/tmp/ansible-tmp-1593672192.41-39427683052579/source", "state": "file", "uid": 0}

TASK [Create a daemonset with privileged access to verify zvol properties at node level] ***
task path: /experiments/functional/zfs-LocalPV/zv-property-runtime-modify/test.yml:25
2020-07-02T06:43:12.628774 (delta: 0.250907)         elapsed: 2.648186 ******** 
changed: [127.0.0.1] => {"changed": true, "cmd": "kubectl create -f ./zv_property_ds.yml", "delta": "0:00:00.893576", "end": "2020-07-02 06:43:13.653877", "rc": 0, "start": "2020-07-02 06:43:12.760301", "stderr": "", "stderr_lines": [], "stdout": "daemonset.apps/zv-property-modify-zfs-w9xrd created", "stdout_lines": ["daemonset.apps/zv-property-modify-zfs-w9xrd created"]}

TASK [Confirm that the ds pods are running on all nodes] ***********************
task path: /experiments/functional/zfs-LocalPV/zv-property-runtime-modify/test.yml:31
2020-07-02T06:43:13.698372 (delta: 1.069562)         elapsed: 3.717784 ******** 
FAILED - RETRYING: Confirm that the ds pods are running on all nodes (20 retries left).
FAILED - RETRYING: Confirm that the ds pods are running on all nodes (19 retries left).
changed: [127.0.0.1] => {"attempts": 3, "changed": true, "cmd": "kubectl get pod -l app=zv-property-modify-zfs --no-headers -o custom-columns=:status.phase | sort | uniq", "delta": "0:00:00.637854", "end": "2020-07-02 06:43:25.982932", "rc": 0, "start": "2020-07-02 06:43:25.345078", "stderr": "", "stderr_lines": [], "stdout": "Running", "stdout_lines": ["Running"]}

TASK [Get the node name on which application is scheduled] *********************
task path: /experiments/functional/zfs-LocalPV/zv-property-runtime-modify/test.yml:42
2020-07-02T06:43:26.010011 (delta: 12.311585)         elapsed: 16.029423 ****** 
changed: [127.0.0.1] => {"changed": true, "cmd": "kubectl get pod -n aman -l app=busybox-sts --no-headers -o custom-columns=:.spec.nodeName", "delta": "0:00:00.570511", "end": "2020-07-02 06:43:26.713617", "rc": 0, "start": "2020-07-02 06:43:26.143106", "stderr": "", "stderr_lines": [], "stdout": "k8s-node2", "stdout_lines": ["k8s-node2"]}

TASK [Get the daemonset pod name which is scheduled on the same node as of application node] ***
task path: /experiments/functional/zfs-LocalPV/zv-property-runtime-modify/test.yml:50
2020-07-02T06:43:26.750915 (delta: 0.740867)         elapsed: 16.770327 ******* 
changed: [127.0.0.1] => {"changed": true, "cmd": "kubectl get pod -l app=zv-property-modify-zfs --no-headers -o jsonpath='{.items[?(@.spec.nodeName==\"k8s-node2\")].metadata.name}'", "delta": "0:00:00.701788", "end": "2020-07-02 06:43:27.602675", "rc": 0, "start": "2020-07-02 06:43:26.900887", "stderr": "", "stderr_lines": [], "stdout": "zv-property-modify-zfs-w9xrd-jbw49", "stdout_lines": ["zv-property-modify-zfs-w9xrd-jbw49"]}

TASK [Record the daemonset pod name scheduled on the same node with application pod] ***
task path: /experiments/functional/zfs-LocalPV/zv-property-runtime-modify/test.yml:58
2020-07-02T06:43:27.627862 (delta: 0.876895)         elapsed: 17.647274 ******* 
ok: [127.0.0.1] => {"ansible_facts": {"ds_pod": "zv-property-modify-zfs-w9xrd-jbw49"}, "changed": false}

TASK [Get the zvolume name] ****************************************************
task path: /experiments/functional/zfs-LocalPV/zv-property-runtime-modify/test.yml:62
2020-07-02T06:43:27.676340 (delta: 0.048415)         elapsed: 17.695752 ******* 
changed: [127.0.0.1] => {"changed": true, "cmd": "kubectl get pvc openebs-busybox -n aman --no-headers -o custom-columns=:.spec.volumeName", "delta": "0:00:00.592873", "end": "2020-07-02 06:43:28.397423", "rc": 0, "start": "2020-07-02 06:43:27.804550", "stderr": "", "stderr_lines": [], "stdout": "pvc-4c615ac1-55a2-4cb8-a183-61ed8b40eb46", "stdout_lines": ["pvc-4c615ac1-55a2-4cb8-a183-61ed8b40eb46"]}

TASK [Record the zvolume name] *************************************************
task path: /experiments/functional/zfs-LocalPV/zv-property-runtime-modify/test.yml:70
2020-07-02T06:43:28.422559 (delta: 0.746168)         elapsed: 18.441971 ******* 
ok: [127.0.0.1] => {"ansible_facts": {"zv_name": "pvc-4c615ac1-55a2-4cb8-a183-61ed8b40eb46"}, "changed": false}

TASK [Get the compression parameter value from the zvolume] ********************
task path: /experiments/functional/zfs-LocalPV/zv-property-runtime-modify/test.yml:74
2020-07-02T06:43:28.463580 (delta: 0.040982)         elapsed: 18.482992 ******* 
changed: [127.0.0.1] => {"changed": true, "cmd": "kubectl get zv pvc-4c615ac1-55a2-4cb8-a183-61ed8b40eb46 -n openebs --no-headers -o custom-columns=:.spec.compression", "delta": "0:00:00.580425", "end": "2020-07-02 06:43:29.168963", "rc": 0, "start": "2020-07-02 06:43:28.588538", "stderr": "", "stderr_lines": [], "stdout": "on", "stdout_lines": ["on"]}

TASK [Get the Dedup parameter value from the zvolume] **************************
task path: /experiments/functional/zfs-LocalPV/zv-property-runtime-modify/test.yml:82
2020-07-02T06:43:29.193584 (delta: 0.729954)         elapsed: 19.212996 ******* 
changed: [127.0.0.1] => {"changed": true, "cmd": "kubectl get zv pvc-4c615ac1-55a2-4cb8-a183-61ed8b40eb46 -n openebs --no-headers -o custom-columns=:.spec.dedup", "delta": "0:00:00.640889", "end": "2020-07-02 06:43:29.958182", "rc": 0, "start": "2020-07-02 06:43:29.317293", "stderr": "", "stderr_lines": [], "stdout": "on", "stdout_lines": ["on"]}

TASK [Get the yaml file for zvolume] *******************************************
task path: /experiments/functional/zfs-LocalPV/zv-property-runtime-modify/test.yml:90
2020-07-02T06:43:29.987568 (delta: 0.793929)         elapsed: 20.00698 ******** 
changed: [127.0.0.1] => {"changed": true, "cmd": "kubectl get zv pvc-4c615ac1-55a2-4cb8-a183-61ed8b40eb46 -n openebs -o yaml > zv.yml", "delta": "0:00:00.614150", "end": "2020-07-02 06:43:30.761187", "rc": 0, "start": "2020-07-02 06:43:30.147037", "stderr": "", "stderr_lines": [], "stdout": "", "stdout_lines": []}

TASK [Modify the compression parameter value] **********************************
task path: /experiments/functional/zfs-LocalPV/zv-property-runtime-modify/test.yml:96
2020-07-02T06:43:30.787867 (delta: 0.800266)         elapsed: 20.807279 ******* 
changed: [127.0.0.1] => {"changed": true, "msg": "1 replacements made"}

TASK [Modify the dedup parameter value] ****************************************
task path: /experiments/functional/zfs-LocalPV/zv-property-runtime-modify/test.yml:102
2020-07-02T06:43:31.079958 (delta: 0.292059)         elapsed: 21.09937 ******** 
changed: [127.0.0.1] => {"changed": true, "msg": "1 replacements made"}

TASK [Apply the modified yaml to update the new value of zvolume parameters] ***
task path: /experiments/functional/zfs-LocalPV/zv-property-runtime-modify/test.yml:108
2020-07-02T06:43:31.251784 (delta: 0.171793)         elapsed: 21.271196 ******* 
changed: [127.0.0.1] => {"changed": true, "cmd": "kubectl apply -f zv.yml", "delta": "0:00:00.686380", "end": "2020-07-02 06:43:32.065050", "failed_when_result": false, "rc": 0, "start": "2020-07-02 06:43:31.378670", "stderr": "", "stderr_lines": [], "stdout": "zfsvolume.zfs.openebs.io/pvc-4c615ac1-55a2-4cb8-a183-61ed8b40eb46 configured", "stdout_lines": ["zfsvolume.zfs.openebs.io/pvc-4c615ac1-55a2-4cb8-a183-61ed8b40eb46 configured"]}

TASK [Verify that compression parameter value is modified in zvolume] **********
task path: /experiments/functional/zfs-LocalPV/zv-property-runtime-modify/test.yml:116
2020-07-02T06:43:32.094980 (delta: 0.843158)         elapsed: 22.114392 ******* 
 [WARNING]: when statements should not include jinja2 templating delimiters
such as {{ }} or {% %}. Found: modified_compress_val.stdout == "{{
new_compress_val }}"
changed: [127.0.0.1] => {"attempts": 1, "changed": true, "cmd": "kubectl get zv pvc-4c615ac1-55a2-4cb8-a183-61ed8b40eb46 -n openebs --no-headers -o custom-columns=:.spec.compression", "delta": "0:00:00.629152", "end": "2020-07-02 06:43:32.846378", "rc": 0, "start": "2020-07-02 06:43:32.217226", "stderr": "", "stderr_lines": [], "stdout": "off", "stdout_lines": ["off"]}

TASK [Verify that compression parameter value is modified in dataset/zvolume on node] ***
task path: /experiments/functional/zfs-LocalPV/zv-property-runtime-modify/test.yml:127
2020-07-02T06:43:32.888919 (delta: 0.793907)         elapsed: 22.908331 ******* 
changed: [127.0.0.1] => {"changed": true, "cmd": "kubectl exec -ti zv-property-modify-zfs-w9xrd-jbw49 -- bash -c 'zfs get all test/pvc-4c615ac1-55a2-4cb8-a183-61ed8b40eb46 | grep compression'", "delta": "0:00:00.775549", "end": "2020-07-02 06:43:33.809660", "failed_when_result": false, "rc": 0, "start": "2020-07-02 06:43:33.034111", "stderr": "Unable to use a TTY - input is not a terminal or the right kind of file", "stderr_lines": ["Unable to use a TTY - input is not a terminal or the right kind of file"], "stdout": "test/pvc-4c615ac1-55a2-4cb8-a183-61ed8b40eb46  compression           off                    local", "stdout_lines": ["test/pvc-4c615ac1-55a2-4cb8-a183-61ed8b40eb46  compression           off                    local"]}

TASK [Verify that dedup parameter value is modified in zvolume] ****************
task path: /experiments/functional/zfs-LocalPV/zv-property-runtime-modify/test.yml:135
2020-07-02T06:43:33.848862 (delta: 0.959912)         elapsed: 23.868274 ******* 
 [WARNING]: when statements should not include jinja2 templating delimiters
such as {{ }} or {% %}. Found: modified_dedup_val.stdout == "{{ new_dedup_val
}}"
changed: [127.0.0.1] => {"attempts": 1, "changed": true, "cmd": "kubectl get zv pvc-4c615ac1-55a2-4cb8-a183-61ed8b40eb46 -n openebs --no-headers -o custom-columns=:.spec.dedup", "delta": "0:00:00.588000", "end": "2020-07-02 06:43:34.567735", "rc": 0, "start": "2020-07-02 06:43:33.979735", "stderr": "", "stderr_lines": [], "stdout": "off", "stdout_lines": ["off"]}

TASK [Verify that compression parameter value is modified in dataset/zvolume on node] ***
task path: /experiments/functional/zfs-LocalPV/zv-property-runtime-modify/test.yml:146
2020-07-02T06:43:34.604297 (delta: 0.755404)         elapsed: 24.623709 ******* 
changed: [127.0.0.1] => {"changed": true, "cmd": "kubectl exec -ti zv-property-modify-zfs-w9xrd-jbw49 -- bash -c 'zfs get all test/pvc-4c615ac1-55a2-4cb8-a183-61ed8b40eb46 | grep dedup'", "delta": "0:00:00.780988", "end": "2020-07-02 06:43:35.513670", "failed_when_result": false, "rc": 0, "start": "2020-07-02 06:43:34.732682", "stderr": "Unable to use a TTY - input is not a terminal or the right kind of file", "stderr_lines": ["Unable to use a TTY - input is not a terminal or the right kind of file"], "stdout": "test/pvc-4c615ac1-55a2-4cb8-a183-61ed8b40eb46  dedup                 off                    local", "stdout_lines": ["test/pvc-4c615ac1-55a2-4cb8-a183-61ed8b40eb46  dedup                 off                    local"]}

TASK [set_fact] ****************************************************************
task path: /experiments/functional/zfs-LocalPV/zv-property-runtime-modify/test.yml:154
2020-07-02T06:43:35.552330 (delta: 0.947986)         elapsed: 25.571742 ******* 
ok: [127.0.0.1] => {"ansible_facts": {"flag": "Pass"}, "changed": false}

TASK [include_tasks] ***********************************************************
task path: /experiments/functional/zfs-LocalPV/zv-property-runtime-modify/test.yml:163
2020-07-02T06:43:35.599637 (delta: 0.047273)         elapsed: 25.619049 ******* 
included: /utils/fcm/update_litmus_result_resource.yml for 127.0.0.1

TASK [Generate the litmus result CR to reflect SOT (Start of Test)] ************
task path: /utils/fcm/update_litmus_result_resource.yml:3
2020-07-02T06:43:35.635531 (delta: 0.035848)         elapsed: 25.654943 ******* 
skipping: [127.0.0.1] => {"changed": false, "skip_reason": "Conditional result was False"}

TASK [Analyze the cr yaml] *****************************************************
task path: /utils/fcm/update_litmus_result_resource.yml:14
2020-07-02T06:43:35.669602 (delta: 0.034029)         elapsed: 25.689014 ******* 
skipping: [127.0.0.1] => {"changed": false, "skip_reason": "Conditional result was False"}

TASK [Apply the litmus result CR] **********************************************
task path: /utils/fcm/update_litmus_result_resource.yml:17
2020-07-02T06:43:35.694598 (delta: 0.024963)         elapsed: 25.71401 ******** 
skipping: [127.0.0.1] => {"changed": false, "skip_reason": "Conditional result was False"}

TASK [Generate the litmus result CR to reflect EOT (End of Test)] **************
task path: /utils/fcm/update_litmus_result_resource.yml:27
2020-07-02T06:43:35.717726 (delta: 0.023065)         elapsed: 25.737138 ******* 
changed: [127.0.0.1] => {"changed": true, "checksum": "0d5cc8334eefedac4d52dfa16bbebe343f71cd44", "dest": "./litmus-result.yaml", "gid": 0, "group": "root", "md5sum": "3d78df64eca1a39b00169e7cd1d23f15", "mode": "0644", "owner": "root", "size": 427, "src": "/root/.ansible/tmp/ansible-tmp-1593672215.76-109329670687609/source", "state": "file", "uid": 0}

TASK [Analyze the cr yaml] *****************************************************
task path: /utils/fcm/update_litmus_result_resource.yml:38
2020-07-02T06:43:36.971487 (delta: 1.253729)         elapsed: 26.990899 ******* 
changed: [127.0.0.1] => {"changed": true, "cmd": "cat litmus-result.yaml", "delta": "0:00:00.715034", "end": "2020-07-02 06:43:37.817791", "rc": 0, "start": "2020-07-02 06:43:37.102757", "stderr": "", "stderr_lines": [], "stdout": "---\napiVersion: litmus.io/v1alpha1\nkind: LitmusResult\nmetadata:\n\n  # name of the litmus testcase\n  name: zvol-property-runtime-modify \nspec:\n\n  # holds information on the testcase\n  testMetadata:\n    app:  \n    chaostype:  \n\n  # holds the state of testcase,  manually updated by json merge patch\n  # result is the useful value today, but anticipate phase use in future \n  testStatus: \n    phase: completed  \n    result: Pass ", "stdout_lines": ["---", "apiVersion: litmus.io/v1alpha1", "kind: LitmusResult", "metadata:", "", "  # name of the litmus testcase", "  name: zvol-property-runtime-modify ", "spec:", "", "  # holds information on the testcase", "  testMetadata:", "    app:  ", "    chaostype:  ", "", "  # holds the state of testcase,  manually updated by json merge patch", "  # result is the useful value today, but anticipate phase use in future ", "  testStatus: ", "    phase: completed  ", "    result: Pass "]}

TASK [Apply the litmus result CR] **********************************************
task path: /utils/fcm/update_litmus_result_resource.yml:41
2020-07-02T06:43:37.840459 (delta: 0.868943)         elapsed: 27.859871 ******* 
changed: [127.0.0.1] => {"changed": true, "failed_when_result": false, "method": "patch", "result": {"apiVersion": "litmus.io/v1alpha1", "kind": "LitmusResult", "metadata": {"creationTimestamp": "2020-07-01T08:23:18Z", "generation": 10, "name": "zvol-property-runtime-modify", "resourceVersion": "952807", "selfLink": "/apis/litmus.io/v1alpha1/litmusresults/zvol-property-runtime-modify", "uid": "609f3a7a-2d29-472e-8502-2a810730f1db"}, "spec": {"testMetadata": {}, "testStatus": {"phase": "completed", "result": "Pass"}}}}
META: ran handlers
META: ran handlers

PLAY RECAP *********************************************************************
127.0.0.1                  : ok=28   changed=22   unreachable=0    failed=0   

2020-07-02T06:43:38.507395 (delta: 0.666908)         elapsed: 28.526807 ******* 
=============================================================================== 
```
